### PR TITLE
added tracelogging to LaunchReporter & TestReporter

### PR DESCRIPTION
**Pull request by vitali-sonchyk-epam**

### CODE REVIEW

**Comment by vitali-sonchyk-epam** [comment]
> ```cs
> Trace.Error(exp);
> ```

Trace.Error(exp.Message); ?

**Comment by vitali-sonchyk-epam** [comment]
> We will not see exception with stacktrace in log.
> 
> I propose to do it like
> 
> ```cs
> var exp = new SomeException("with message");
> Trace.Error(exp);
> throw exp;
> ```

fixed

File: src/ReportPortal.Shared/Reporter/LaunchReporter.cs [hunk 1]
```diff
@@ -98,12 +104,16 @@ public void Finish(FinishLaunchRequest request)
                 {
                     if (StartTask.IsFaulted)
                     {
-                        throw new Exception("Cannot finish launch due starting launch failed.", StartTask.Exception);
+                        var exp = new Exception("Cannot finish launch due starting launch failed.");
```
Comment (nvborisenko) [inline src/ReportPortal.Shared/Reporter/LaunchReporter.cs]: Loosing inner exception.

File: src/ReportPortal.Shared/Reporter/TestReporter.cs [hunk 2]
```diff
@@ -107,18 +115,22 @@ public void Finish(FinishTestItemRequest request)
...
 
-            FinishTask = Task.Factory.ContinueWhenAll(dependentTasks.ToArray(), async (a) =>
+            FinishTask = Task.Factory.ContinueWhenAll(dependentTasks.ToArray(), async a =>
             {
                 try
                 {
                     if (StartTask.IsFaulted)
                     {
-                        throw new Exception("Cannot finish test item due starting item failed.", StartTask.Exception);
+                        var exp = new Exception("Cannot finish test item due starting item failed.");
```
Comment (nvborisenko) [inline src/ReportPortal.Shared/Reporter/TestReporter.cs]: Loosing inner exception.

**Pull request merged**